h2o-core/src/main/java/water/parser/ParseSetup.java

```diff
@@ -346,13 +357,14 @@ public GuessSetupTsk(ParseSetup userSetup) {
         _maxLineLength = maxLineLength(bits);
         if (_maxLineLength==-1) throw new H2OIllegalArgumentException("The first 4MB of the data don't contain any line breaks. Cannot parse.");
```

Comment:
this will never be satisfied because you changed maxLineLength to return 0 instead of -1